File: VB/WinSolution.Module.Win/ResetSettingsViewController.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.ExpressApp
Imports DevExpress.XtraEditors
Imports DevExpress.ExpressApp.Win.Layout
Imports DevExpress.ExpressApp.Model.Core

Namespace WinSolution.Module.Win
	Partial Public Class ResetSettingsViewController
		Inherits ViewController
		Private layoutManager As WinLayoutManager
		Public Sub New()
			InitializeComponent()
			RegisterActions(components)
		End Sub
		Protected Overrides Overloads Sub OnViewChanging(ByVal view As View)
			MyBase.OnViewChanging(view)
			Dim dv As DetailView = TryCast(view, DetailView)
			If dv IsNot Nothing Then
				layoutManager = TryCast(dv.LayoutManager, WinLayoutManager)
				Active("DetailViewUsesWinLayoutManager") = layoutManager IsNot Nothing
			End If
		End Sub
		Private Sub saResetSettings_Execute(ByVal sender As Object, ByVal e As DevExpress.ExpressApp.Actions.SimpleActionExecuteEventArgs) Handles saResetSettings.Execute
			Try
				If TypeOf View Is DetailView Then
					layoutManager.Container.RestoreDefaultLayout()
				Else
                    CType(View.Model, ModelNode).Undo()
                    View.SynchronizeWithInfo()
				End If
			Catch
				XtraMessageBox.Show("An error occurred when resetting the View's settings. Please contact your application administrator for a solution. We will try to close the current View after you press the OK button.", "Reset Settings", System.Windows.Forms.MessageBoxButtons.OK, System.Windows.Forms.MessageBoxIcon.Error)
				View.Close()
			End Try
		End Sub
	End Class
End Namespace

File: CS/E1897/WinSolution.Module.Win/ResetSettingsViewController.cs
using System;
using DevExpress.ExpressApp;
using DevExpress.XtraEditors;
using DevExpress.ExpressApp.Win.Layout;
using DevExpress.ExpressApp.Model.Core;

namespace WinSolution.Module.Win {
    public partial class ResetSettingsViewController : ViewController {
        private WinLayoutManager layoutManager;
        public ResetSettingsViewController() {
            InitializeComponent();
            RegisterActions(components);
        }
        protected override void OnViewChanging(View view) {
            base.OnViewChanging(view);
            DetailView dv = view as DetailView;
            if (dv != null) {
                layoutManager = dv.LayoutManager as WinLayoutManager;
                Active["DetailViewUsesWinLayoutManager"] = layoutManager != null;
            }
        }
        private void saResetSettings_Execute(object sender, DevExpress.ExpressApp.Actions.SimpleActionExecuteEventArgs e) {
            try {
                if (View is DetailView) {
                    layoutManager.Container.RestoreDefaultLayout();
                }
                else {
                    ((ModelNode)View.Model).Undo();
                    View.SynchronizeWithInfo();
                }
            } catch {
                XtraMessageBox.Show("An error occurred when resetting the View's settings. Please contact your application administrator for a solution. We will try to close the current View after you press the OK button.", "Reset Settings", System.Windows.Forms.MessageBoxButtons.OK, System.Windows.Forms.MessageBoxIcon.Error);
                View.Close();
            }
        }
    }
}